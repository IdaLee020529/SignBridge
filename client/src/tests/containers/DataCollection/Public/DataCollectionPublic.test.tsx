import { create } from "react-test-renderer";
import { render, fireEvent, cleanup, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom"; //For extra validation => toBeInTheDocument()
import DataCollectionPublic from "../../../../containers/DataCollection/Public/DataCollectionPublic";

jest.mock("react-i18next", () => ({
  useTranslation: () => ({
    t: (key: string) => key, // Mock the translation function to return the key
    i18n: {
      changeLanguage: jest.fn(), // Mock the changeLanguage function
    },
  }),
}));

jest.mock("js-cookie");

describe("Test DataCollection", () => {
  afterEach(() => {
    // Clean up the DOM
    cleanup();
    // Reset mocks to their initial state
    jest.resetAllMocks();
  });
  it("should render correctly", () => {
    const tree = create(<DataCollectionPublic />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should be able to submit", async () => {
    const mockFormData = new FormData();
    mockFormData.append("user_id", "mock_user_id");
    mockFormData.append("name", "John Doe");
    mockFormData.append("email", "john@example.com");
    mockFormData.append("text_sentence", "Test sentence");
    mockFormData.append("status_SE", "New");
    mockFormData.append("status_Admin", "-");

    jest.spyOn(require("js-cookie"), "get").mockReturnValueOnce("valid_token"); // Simulate logged-in user

    // Mocking the selected video file => For video upload
    Object.defineProperty(global, "File", {
      value: class extends Blob {
        constructor(
          fileParts: Array<string>,
          fileName: string,
          options?: BlobPropertyBag
        ) {
          super(fileParts, options);
          (this as any).name = fileName;
        }
      },
    });

    //Use asFragment to render toMatchSnapshot
    const { asFragment } = render(<DataCollectionPublic />);

    // Fill in the form fields
    userEvent.type(screen.getByTestId("name"), "John Doe"); //If u go to my code u will find that i have put up testid for it its easier to use
    userEvent.type(screen.getByTestId("email"), "john@example.com");
    userEvent.type(screen.getByTestId("text"), "Test sentence");

    // Mock the Upload component behavior
    jest.mock("antd", () => ({
      Upload: ({ children }: { children: React.ReactNode }) => (
        <div data-testid="upload-mock">{children}</div>
      ),
    }));
    // const uploadButton = await screen.findByTestId("UploadVideo"); // Wait for the button to be available

    // fireEvent.click(uploadButton); // Click the button

    // Mock the customRequest function to simulate file upload
    fireEvent.click(screen.getByText("submit_btn"));

    //Using snapshot as temporary solution
    expect(asFragment()).toMatchSnapshot();
    // expect(submitForm).toHaveBeenCalledWith(mockFormData);
    //Originally my expected output: But i have implemented the modal
    // expect(screen.getByTestId("modal")).toBeInTheDocument();
  });
});
