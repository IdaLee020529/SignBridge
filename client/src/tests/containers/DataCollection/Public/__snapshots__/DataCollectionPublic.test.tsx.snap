// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DataCollection should be able to submit 1`] = `
<DocumentFragment>
  <div>
    <div
      class="dataForm-bg"
    >
      <div
        class="dataSubmissionForm"
      >
        <div
          class="dataForm"
        >
          <div
            class="dataForm-header-container"
          >
            <div
              class="dataForm-header"
            >
              <h1>
                dataset_collection_form
              </h1>
            </div>
          </div>
          <div
            class="dataForm-cover"
          >
            <div
              class="dataForm-card public"
            >
              <h1>
                in_touch
              </h1>
              <h3>
                more_suggestions
              </h3>
              <div
                class="dataForm-card-content"
              >
                <div
                  class="dataForm-card-info"
                >
                  <svg
                    fill="currentColor"
                    height="2em"
                    stroke="currentColor"
                    stroke-width="0"
                    style="color: rgb(0, 0, 0); margin-right: 5px;"
                    viewBox="0 0 24 24"
                    width="2em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M12 2C8.13 2 5 5.13 5 9c0 1.74.5 3.37 1.41 4.84.95 1.54 2.2 2.86 3.16 4.4.47.75.81 1.45 1.17 2.26.26.55.47 1.5 1.26 1.5s1-.95 1.25-1.5c.37-.81.7-1.51 1.17-2.26.96-1.53 2.21-2.85 3.16-4.4C18.5 12.37 19 10.74 19 9c0-3.87-3.13-7-7-7zm0 9.75a2.5 2.5 0 0 1 0-5 2.5 2.5 0 0 1 0 5z"
                    />
                  </svg>
                  <p>
                     neoun_address
                  </p>
                </div>
                <div
                  class="dataForm-card-info2"
                >
                  <svg
                    fill="currentColor"
                    height="2em"
                    stroke="currentColor"
                    stroke-width="0"
                    style="color: rgb(0, 0, 0); margin-right: 5px;"
                    viewBox="0 0 24 24"
                    width="2em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M6.62 10.79c1.44 2.83 3.76 5.14 6.59 6.59l2.2-2.2c.27-.27.67-.36 1.02-.24 1.12.37 2.33.57 3.57.57.55 0 1 .45 1 1V20c0 .55-.45 1-1 1-9.39 0-17-7.61-17-17 0-.55.45-1 1-1h3.5c.55 0 1 .45 1 1 0 1.25.2 2.45.57 3.57.11.35.03.74-.25 1.02l-2.2 2.2z"
                    />
                  </svg>
                  <p>
                    neoun_phone
                  </p>
                </div>
                <div
                  class="dataForm-card-info2"
                >
                  <svg
                    fill="currentColor"
                    height="30px"
                    stroke="currentColor"
                    stroke-width="0"
                    style="color: rgb(0, 0, 0); margin-right: 5px;"
                    viewBox="0 0 24 24"
                    width="30px"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
                    />
                  </svg>
                  <p>
                    neoun_email
                  </p>
                </div>
                <div
                  class="dataForm-card-info"
                />
              </div>
            </div>
            <div
              class="dataForm-container"
            >
              <div
                class="dataForm-info-block"
              >
                <svg
                  class="dataForm-info"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  style="color: rgb(0, 0, 0);"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3 4v16a1 1 0 0 0 1 1h16a1 1 0 0 0 1-1V4a1 1 0 0 0-1-1H4a1 1 0 0 0-1 1zm8 3h2v2h-2V7zm0 4h2v6h-2v-6z"
                  />
                </svg>
              </div>
              <div
                class="row justify-content-center input-container"
              >
                <div
                  class="col-md-8"
                >
                  <form
                    novalidate=""
                  >
                    <div
                      class="form-group "
                    >
                      <input
                        class="form-control "
                        data-testid="name"
                        name="name"
                        placeholder=" "
                        type="text"
                        value=""
                      />
                      <label
                        class="input-label "
                      >
                        name_input:
                      </label>
                      <div
                        class="form-error "
                      />
                    </div>
                    <div
                      class="form-group "
                    >
                      <input
                        class="form-control "
                        data-testid="email"
                        name="email"
                        placeholder=" "
                        type="email"
                        value=""
                      />
                      <label
                        class="input-label "
                      >
                        email_input:
                      </label>
                      <div
                        class="form-error "
                      />
                    </div>
                    <div
                      class="form-group "
                    >
                      <textarea
                        class="form-control  multipleLines"
                        data-testid="text"
                        name="text"
                        placeholder=" "
                      />
                      <label
                        class="input-label multiple_label"
                      >
                        text_sentence_input:
                      </label>
                      <div
                        class="form-error "
                      />
                    </div>
                    <div
                      class="video-container"
                    >
                      <div
                        class="videoinput-class"
                      >
                        <span
                          class="ant-upload-wrapper css-dev-only-do-not-override-42nv3w"
                        >
                          <div
                            class="ant-upload ant-upload-select"
                          >
                            <span
                              class="ant-upload"
                            >
                              <input
                                accept=".mp4"
                                style="display: none;"
                                type="file"
                              />
                              <button
                                class="ant-btn css-dev-only-do-not-override-42nv3w ant-btn-default ant-btn-lg"
                                type="button"
                              >
                                <span
                                  class="ant-btn-icon"
                                >
                                  <span
                                    aria-label="upload"
                                    class="anticon anticon-upload"
                                    role="img"
                                  >
                                    <svg
                                      aria-hidden="true"
                                      data-icon="upload"
                                      fill="currentColor"
                                      focusable="false"
                                      height="1em"
                                      viewBox="64 64 896 896"
                                      width="1em"
                                    >
                                      <path
                                        d="M400 317.7h73.9V656c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V317.7H624c6.7 0 10.4-7.7 6.3-12.9L518.3 163a8 8 0 00-12.6 0l-112 141.7c-4.1 5.3-.4 13 6.3 13zM878 626h-60c-4.4 0-8 3.6-8 8v154H214V634c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v198c0 17.7 14.3 32 32 32h684c17.7 0 32-14.3 32-32V634c0-4.4-3.6-8-8-8z"
                                      />
                                    </svg>
                                  </span>
                                </span>
                                <span>
                                  choose_a_video
                                </span>
                              </button>
                            </span>
                          </div>
                        </span>
                      </div>
                    </div>
                    <div
                      class="button-container"
                    >
                      <button
                        class="btn btn--reset btn--large"
                        type="button"
                      >
                        reset_btn
                      </button>
                      <button
                        class="btn btn--submit btn--large"
                        type="submit"
                      >
                        submit_btn
                      </button>
                    </div>
                  </form>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Test DataCollection should render correctly 1`] = `
<div>
  <div
    className="dataForm-bg"
  >
    <div
      className="dataSubmissionForm"
    >
      <div
        className="dataForm"
      >
        <div
          className="dataForm-header-container"
        >
          <div
            className="dataForm-header"
          >
            <h1>
              dataset_collection_form
            </h1>
          </div>
        </div>
        <div
          className="dataForm-cover"
        >
          <div
            className="dataForm-card public"
          >
            <h1>
              in_touch
            </h1>
            <h3>
              more_suggestions
            </h3>
            <div
              className="dataForm-card-content"
            >
              <div
                className="dataForm-card-info"
              >
                <svg
                  fill="currentColor"
                  height="2em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    {
                      "color": "#000",
                      "marginRight": "5px",
                    }
                  }
                  viewBox="0 0 24 24"
                  width="2em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M12 2C8.13 2 5 5.13 5 9c0 1.74.5 3.37 1.41 4.84.95 1.54 2.2 2.86 3.16 4.4.47.75.81 1.45 1.17 2.26.26.55.47 1.5 1.26 1.5s1-.95 1.25-1.5c.37-.81.7-1.51 1.17-2.26.96-1.53 2.21-2.85 3.16-4.4C18.5 12.37 19 10.74 19 9c0-3.87-3.13-7-7-7zm0 9.75a2.5 2.5 0 0 1 0-5 2.5 2.5 0 0 1 0 5z"
                  />
                </svg>
                <p>
                   
                  neoun_address
                </p>
              </div>
              <div
                className="dataForm-card-info2"
              >
                <svg
                  fill="currentColor"
                  height="2em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    {
                      "color": "#000",
                      "marginRight": "5px",
                    }
                  }
                  viewBox="0 0 24 24"
                  width="2em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M6.62 10.79c1.44 2.83 3.76 5.14 6.59 6.59l2.2-2.2c.27-.27.67-.36 1.02-.24 1.12.37 2.33.57 3.57.57.55 0 1 .45 1 1V20c0 .55-.45 1-1 1-9.39 0-17-7.61-17-17 0-.55.45-1 1-1h3.5c.55 0 1 .45 1 1 0 1.25.2 2.45.57 3.57.11.35.03.74-.25 1.02l-2.2 2.2z"
                  />
                </svg>
                <p>
                  neoun_phone
                </p>
              </div>
              <div
                className="dataForm-card-info2"
              >
                <svg
                  fill="currentColor"
                  height="30px"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    {
                      "color": "#000",
                      "marginRight": "5px",
                    }
                  }
                  viewBox="0 0 24 24"
                  width="30px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
                  />
                </svg>
                <p>
                  neoun_email
                </p>
              </div>
              <div
                className="dataForm-card-info"
              />
            </div>
          </div>
          <div
            className="dataForm-container"
          >
            <div
              className="dataForm-info-block"
            >
              <svg
                className="dataForm-info"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  {
                    "color": "#000",
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3 4v16a1 1 0 0 0 1 1h16a1 1 0 0 0 1-1V4a1 1 0 0 0-1-1H4a1 1 0 0 0-1 1zm8 3h2v2h-2V7zm0 4h2v6h-2v-6z"
                />
              </svg>
            </div>
            <div
              className="row justify-content-center input-container"
            >
              <div
                className="col-md-8"
              >
                <form
                  noValidate={true}
                  onSubmit={[Function]}
                >
                  <div
                    className="form-group "
                  >
                    <input
                      className="form-control "
                      data-testid="name"
                      name="name"
                      onChange={[Function]}
                      placeholder=" "
                      type="text"
                      value=""
                    />
                    <label
                      className="input-label "
                    >
                      name_input
                      :
                    </label>
                    <div
                      className="form-error "
                    />
                  </div>
                  <div
                    className="form-group "
                  >
                    <input
                      className="form-control "
                      data-testid="email"
                      name="email"
                      onChange={[Function]}
                      placeholder=" "
                      type="email"
                      value=""
                    />
                    <label
                      className="input-label "
                    >
                      email_input
                      :
                    </label>
                    <div
                      className="form-error "
                    />
                  </div>
                  <div
                    className="form-group "
                  >
                    <textarea
                      className="form-control  multipleLines"
                      data-testid="text"
                      name="text"
                      onChange={[Function]}
                      placeholder=" "
                      value=""
                    />
                    <label
                      className="input-label multiple_label"
                    >
                      text_sentence_input
                      :
                    </label>
                    <div
                      className="form-error "
                    />
                  </div>
                  <div
                    className="video-container"
                  >
                    <div
                      className="videoinput-class"
                    >
                      <span
                        className="ant-upload-wrapper css-dev-only-do-not-override-42nv3w"
                      >
                        <div
                          className="ant-upload ant-upload-select"
                        >
                          <span
                            className="ant-upload"
                            onClick={[Function]}
                            onDragOver={[Function]}
                            onDrop={[Function]}
                            onKeyDown={[Function]}
                          >
                            <input
                              accept=".mp4"
                              disabled={false}
                              multiple={false}
                              onChange={[Function]}
                              onClick={[Function]}
                              style={
                                {
                                  "display": "none",
                                }
                              }
                              type="file"
                            />
                            <button
                              className="ant-btn css-dev-only-do-not-override-42nv3w ant-btn-default ant-btn-lg"
                              disabled={false}
                              onClick={[Function]}
                              style={{}}
                              type="button"
                            >
                              <span
                                className="ant-btn-icon"
                                style={{}}
                              >
                                <span
                                  aria-label="upload"
                                  className="anticon anticon-upload"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="upload"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M400 317.7h73.9V656c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V317.7H624c6.7 0 10.4-7.7 6.3-12.9L518.3 163a8 8 0 00-12.6 0l-112 141.7c-4.1 5.3-.4 13 6.3 13zM878 626h-60c-4.4 0-8 3.6-8 8v154H214V634c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v198c0 17.7 14.3 32 32 32h684c17.7 0 32-14.3 32-32V634c0-4.4-3.6-8-8-8z"
                                    />
                                  </svg>
                                </span>
                              </span>
                              <span>
                                choose_a_video
                              </span>
                            </button>
                          </span>
                        </div>
                      </span>
                    </div>
                  </div>
                  <div
                    className="button-container"
                  >
                    <button
                      className="btn btn--reset btn--large"
                      onClick={[Function]}
                      type="button"
                    >
                      reset_btn
                    </button>
                    <button
                      className="btn btn--submit btn--large"
                      type="submit"
                    >
                      submit_btn
                    </button>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div />
  </div>
</div>
`;
